libraries/RC_Channel/RC_Channels.cpp

```diff
@@ -345,6 +351,71 @@ RC_Channel &RC_Channels::get_lateral_channel() const
 #endif  // AP_RCMAPPER_ENABLED
 
 
+/*
+  check for pilot input on rudder stick for arming/disarming
+*/
+void RC_Channels::rudder_arm_disarm_check()
+{
+    // run no more code if arm/disarm via rudder input channel is
+    // completely disabled.  Further checks using this parameter are
+    // done below.
+    if (AP::arming().get_rudder_arming_type() == AP_Arming::RudderArming::IS_DISABLED) {
+        return;
+    }
+
+    const RC_Channel *channel = get_arming_channel();
+    if (channel == nullptr) {
+        return;
+    }
+
+    const auto control_in = channel->get_control_in();
+    const auto abs_control_in = abs(control_in);
+
+    if (abs_control_in == 0) {
+        have_seen_neutral_rudder = true;
+    }
+
+    if (abs_control_in <= 4000) {
+        // not trying to (or no longer trying to) arm or disarm
+        rudder_arm_timer = 0;
+        return;
+    }
+
+    // enforce correct stick gesture for arming (but not disarming):
+    if (arming_check_throttle() && control_in > 4000) {
+        // only permit arming if the vehicle isn't being commanded to
+        // move via RC input
+        const auto &c = rc().get_throttle_channel();
+        if (c.get_control_in() != 0) {
```

Comment:
don't you want to use the deadzone version here?

Reply:
I don't think so - this is pilot stick input we want to be inactive.  It would be really, really difficult for a Rover user to get zero input on a mid-stick-is-zero throttle input without a dz :-)